### PR TITLE
Support functions and actions taking/returning delta payloads

### LINKED ISSUE #348: Allow functions/actions to return (and take) a delta payload
We define a delta payload that can be returned from a(n opaque) delta link, as well as other service-defined mechanisms.

However, we don't have a way to define a custom function (or action) that can return a delta payload, and there are many places where a service may want to describe the results of a custom function (or action) using a delta payload.

Similarly, one can imagine using the delta payload as a payload to a custom action.

How can users define functions/actions that can return (or take) delta payloads.

Imported from [ODATA-1527](https://issues.oasis-open.org/browse/ODATA-1527)

### PR DESCRIPTION
**Pull request by mikepizzo**
Fixes #348.
Add description of Core.IsDelta to Parameter and ReturnType
Add, to introduction, that functions and actions can take, and return, delta payloads

Also:
-added missing description of Core.OptionalParameter to Parameter
-fixed a typo ("Action of Function"=>"Action or Function")

### CODE REVIEW

File: odata-csdl/12 Action and Function.md [hunk 1]
```diff
@@ -309,6 +318,15 @@ For collection-valued return types the value of `Type` is the character
...
 type, followed by a closing parenthesis `)`.
 
+### ##subisec Annotation `Core.IsDelta`
+
+The `edm:ReturnType` element of a function or action that returns results as a delta payload
+is annotated with the term [`Core.IsDelta`]($$$OData-VocCore$$$#IsDelta).
+
+Delta payloads represent changes between two versions of data and, in addition
+to current values, MAY include deleted entries as well as changes to related 
+entities and relationships, according to the format-specific delta representation.
```
Comment (HeikoTheissen) [inline odata-csdl/12 Action and Function.md:328]: Can we avoid repeating this sentence?

File: odata-csdl/12 Action and Function.md [hunk 2]
```diff
@@ -271,6 +271,15 @@ present with the literal value `true`.
 
 Absence of the `$Type` member means the type is `Edm.String`.
 
+### ##subisec Annotation `Core.IsDelta`
+
+The `$ReturnType` of a function or action that returns results as a delta payload
+is annotated with the term [`Core.IsDelta`]($$$OData-VocCore$$$#IsDelta).
```
Comment (HeikoTheissen) [inline odata-csdl/12 Action and Function.md:277]: What can the return type be in this case?
* a (collection of an) entity type
* a (collection of a) complex type

Analogous question for parameters

**Comment by ralfhandl** [comment]
Should we remove the "Response" from section headers
* [10.17 Delta Payload Response](https://docs.oasis-open.org/odata/odata/v4.02/csd01/part1-protocol/odata-v4.02-csd01-part1-protocol.html#DeltaPayloadResponse)
* [10.18 Item in a Delta Payload Response](https://docs.oasis-open.org/odata/odata/v4.02/csd01/part1-protocol/odata-v4.02-csd01-part1-protocol.html#IteminaDeltaPayloadResponse)

File: odata-csdl/12 Action and Function.md (hunk 2 above)
Comment (ralfhandl) [reply odata-csdl/12 Action and Function.md:277]: Only entity or collection of entity according to Protocol and JSON Format.

File: odata-csdl/12 Action and Function.md [hunk 3]
```diff
@@ -383,6 +392,15 @@ present with the literal value `true`.
 
 Absence of the `$Type` member means the type is `Edm.String`.
 
+### ##subisec Annotation `Core.IsDelta`
+
+A `$Parameter` that accepts a delta payload
+is annotated with the term [`Core.IsDelta`]($$$OData-VocCore$$$#IsDelta).
```
Comment (HeikoTheissen) [inline odata-csdl/12 Action and Function.md:398]: Why not make this representation-independent as well?

**Review by HeikoTheissen — COMMENTED** [review]
Make the section "Annotation `Core.OptionalParameter`" representation-independent as well.

**Comment by ralfhandl** [comment]
> Make the section "Annotation `Core.OptionalParameter`" representation-independent as well.

The section contains an annotation example which is representation-specific.

**Review by HeikoTheissen — APPROVED** [review]

**Pull request merged**